### PR TITLE
CalibTracker: cleanup clang warnings

### PR DESCRIPTION
**Pull request by gartung**
Cleanup these clang warnings in CalibTracker packages:
CalibTracker/SiPixelTools/plugins/SiPixelFedFillerWordEventNumber.cc:63:16: warning: equality comparison with extraneous parentheses [-Wparentheses-equality]
CalibTracker/SiPixelTools/plugins/SiPixelErrorsDigisToCalibDigis.cc:149:13: warning: reference cannot be bound to dereferenced null pointer in well-defined C++ code; pointer may be assumed to always convert to true [-Wundefined-bool-conversion]
CalibTracker/SiStripChannelGain/interface/SiStripGainsPCLWorker.h:81:18: warning: 'beginJob' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEffFromCalibTree.cc:98:18: warning: 'SiStripHitEffFromCalibTree::algoBeginJob' hides overloaded virtual function [-Woverloaded-virtual]

### CODE REVIEW

**Comment by gartung** [comment]
@Dr15Jones

File: CalibTracker/SiStripChannelGain/interface/SiStripGainsPCLWorker.h [hunk 1]
```diff
@@ -78,11 +78,11 @@ class SiStripGainsPCLWorker : public  DQMEDAnalyzer {
     static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
 
 private:
-    virtual void beginJob() ;
+    virtual void beginJob() override;
     void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;   
     void dqmBeginRun(edm::Run const&, edm::EventSetup const&) override;
     void analyze(const edm::Event&, const edm::EventSetup&) override;
-    virtual void endJob() ;
+    virtual void endJob() override;
```
Comment (Dr15Jones) [inline CalibTracker/SiStripChannelGain/interface/SiStripGainsPCLWorker.h]: `virtual` can be removed from this line.

**Comment by Dr15Jones** [comment]
please test

**Comment by lpernie** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**